tests/algorithms/sbi/test_smcabc.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+import torch
+
+import sbibm
+from sbibm.algorithms.sbi.smcabc import run as run_posterior
+from sbibm.metrics.c2st import c2st
+
+
+@pytest.mark.parametrize(
+    "task_name",
+    [
+        task_name
+        for task_name in [
+            "gaussian_linear",
+            "two_moons",
+        ]
+    ],
+)
+def test_run_posterior_interface(
+    task_name,
+    num_simulations=100,
+    num_samples=5,
+):
+    task = sbibm.get_task(task_name)
+
+    samples, _, _ = run_posterior(
+        task=task,
+        num_simulations=num_simulations,
+        num_observation=3,
+        num_samples=num_samples,
+    )
+
+    # we are not interested in testing for correctness
+    assert len(samples.shape) > 0
```

Comment:
maybe test for number of samples here using `sample.shape == torch.Size(num_samples, task.dim_parameters)`?

Reply:
And squashed another set of bugs! Done with cdd0439